libs/constrained-generators/src/Constrained/Examples/Map.hs

```diff
@@ -80,3 +80,40 @@ lookupSpecific = constrained' $ \k v m ->
   [ m `dependsOn` k
   , assert $ lookup_ k m ==. cJust_ v
   ]
+
+mapRestrictedValues :: Specification BaseFn (Map (Either Int ()) Int)
+mapRestrictedValues = constrained $ \m ->
+  [ assert $ sizeOf_ (dom_ m) ==. 6
+  , forAll' m $ \k v ->
+      caseOn
+        k
+        (branch $ \_ -> 20 <=. v)
+        (branch $ \_ -> True)
+  ]
+
+-- NOTE: this fails if you pick the values of the map first - you're unlikely to generate
+-- three values such that two of them are <= -100 and >= 100 respectively even though
+-- you take satisfiability of the whole elem constraint into account. This can't be fixed
+-- with a `dependsOn v k` because the issue is that we've generated a bunch of values
+-- before we ever go to generate the keys.
```

Comment:
We will have to change genFromTypeSpec @(Map k v), to be more general to make this work, not always generating the values first.